### DISCUSSION TITLE
Foundation.NSErrorException on IOS

### DISCUSSION THREAD

**Discussion by crashnnburn** [dimonovdd/Xamarin.MediaGallery#56] [Q&A] [answered]
Appears on: IOS Platform

Anyone know why I keep getting this error?
When debugging it happens exactly on the line: "await result.OpenReadAsync()" and visually it's after I tap the add button on the gallery view. It does sometimes go through and get's the photos, but most of the time I get this error.

I don't know if this is a simulator issue or visual studio issue since the only other option I have is an old iPhone 6.  It works well but since it's <14.0 and makes it single selection, I can't figure out if this error is because of the simulator or a bug in the package. 

This is the Exception Error I get:
`Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type public.jpeg" UserInfo={NSLocalizedDescription=Cannot load representation of type public.jpeg, NSUnderlyingError=0x60000043bf60 {Error Domain=NSCocoaErrorDomain Code=260 "The file “version=1&uuid=CC95F08C-88C3-4012-9D6D-64A413D254B3&mode=compatible.jpeg” couldn’t be opened because there is no such file." UserInfo={NSURL=file:///Users/eduardo/Library/Developer/CoreSimulator/Devices/32C7531C-708E-4F07-8060-BF3E464A8D30/data/Containers/Shared/AppGroup/B886609C-AA0F-4141-ADA0-A5BD7B682CEF/File%20Provider%20Storage/photospicker/version=1&uuid=CC95F08C-88C3-4012-9D6D-64A413D254B3&mode=compatible.jpeg, NSFilePath=/Users/eduardo/Library/Developer/CoreSimulator/Devices/32C7531C-708E-4F07-8060-BF3E464A8D30/data/Containers/Shared/AppGroup/B886609C-AA0F-4141-ADA0-A5BD7B682CEF/File Provider Storage/photospicker/version=1&uuid=CC95F08C-88C3-4012-9D6D-64A413D254B3&mode=compatible.jpeg, NSUnderlyingError=0x6000004cd8f0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}}}`

**Comment by dimonovdd** [comment, marked as answer]
Hi, I also got this error on the simulator. Try to remove all the default images on the simulator and add your own images or you can download images from safari on the simulator

**Reply by crashnnburn** [reply]
Yup, it seems to have been that. It's doing it way less. Thanks for the suggestion! Appreciate the fast response.
